### PR TITLE
Plugin dependency

### PR DESCRIPTION
**Pull request by bpinto**
@derekstavis could you confirm this works with `nvm`?

### CODE REVIEW

**Comment by derekstavis** [comment]
Awesome! I love it! Testing right now!

**Comment by derekstavis** [comment]
It works amazingly well. While working on #8 we could also include automagic dependency install/loading too, as this PR just supports imperative dependencies, instead of a declarative ones.

**Comment by bpinto** [comment]
True... Perhaps we should delay this PR until we add support for that? I think the way to use this function will be quite different.

**Comment by bpinto** [comment]
I will work on some other things that I feel are necessary before I can dive on this. :)

**Comment by derekstavis** [comment]
Nice! I think that we could wait to tackle dependencies in a row!

**Comment by derekstavis** [comment]
Also, why do this PR addresses `omf destroy`?

**Comment by bpinto** [comment]
You are too pedantic. :)

File: lib/autoload.fish [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 
 function autoload -d "autoload a function or completion path"
+  set -l path_exist "false"
+
   for path in $argv
+    set -l dest fish_function_path
+
     if test -d "$path"
-      set -l dest fish_function_path
+      set path_exist "true"
```
Comment (derekstavis) [inline lib/autoload.fish]: You can just `set -l path_exist` and in the end `set -q path_exist; and return 0; or return 1`.

Comment (bpinto) [inline lib/autoload.fish]: Better!

File: lib/available.fish [hunk 2]
```diff
@@ -4,6 +4,6 @@
 # OVERVIEW
 #   Check if a program is available.
 
-function available -a program -d "check if a program is available."
-  type "$program" ^/dev/null >&2
+function available -a name -d "Check if a program is available."
```
Comment (derekstavis) [inline lib/available.fish]: Leveraging this change, `Check if a function or program is available` describes the command better.

File: lib/require.fish [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#     - Emit its initialization event.
+#
+#   If the required plugin has already been loaded, does nothing.
+#   If the required plugin does not exist, return an error code and send a message to stderr.
+
+function require -a name
+  # Skip if plugin has already been loaded.
+  echo $fish_function_path | grep -q /pkg/$name; and return 0
+
+  set -l plugin_exist "false"
```
Comment (derekstavis) [inline lib/require.fish]: you can apply the same idea of `set -q` here.

**Comment by derekstavis** [comment]
Great!! Cleaner code!! :star2:

**Comment by bpinto** [comment]
Closing as this is not complete. We should work towards having a `bundle` file per plugin.

**Pull request closed without merging**